src/main/java/com/openshift/jenkins/plugins/pipeline/model/IOpenShiftExec.java

```diff
@@ -0,0 +1,126 @@
+package com.openshift.jenkins.plugins.pipeline.model;
+
+import com.openshift.jenkins.plugins.pipeline.Argument;
+import com.openshift.jenkins.plugins.pipeline.MessageConstants;
+import com.openshift.restclient.IClient;
+import com.openshift.restclient.ResourceKind;
+import com.openshift.restclient.api.capabilities.IPodExec;
+import com.openshift.restclient.capability.CapabilityVisitor;
+import com.openshift.restclient.capability.IStoppable;
+import com.openshift.restclient.model.IPod;
+import hudson.Launcher;
+import hudson.model.TaskListener;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+public interface IOpenShiftExec extends IOpenShiftPlugin {
+
+	String DISPLAY_NAME = "OpenShift Exec";
+
+	String getPod();
+
+	String getContainer();
+
+	String getCommand();
+
+	List<Argument> getArguments();
+
+	String getWaitTime();
+
+	default String getDisplayName() {
+		return DISPLAY_NAME;
+	}
+
+	String getWaitTime(Map<String, String> overrides);
+
+	default boolean coreLogic(Launcher launcher, TaskListener listener, Map<String, String> overrides) {
+		listener.getLogger().println(String.format(MessageConstants.START_EXEC, DISPLAY_NAME, getNamespace(overrides)));
+		List<String> fullCommand = new ArrayList<>();
+		fullCommand.add(getCommand());
+		getArguments().forEach( a -> fullCommand.add( a.getValue() ) );
+
+		IPodExec.Options options = new IPodExec.Options();
+		if ( !getContainer().isEmpty() ) {
+			options.container( getContainer() );
+		}
+
+		// get oc client
+		IClient client = this.getClient(listener, DISPLAY_NAME, overrides);
+		if ( client == null ) {
+			listener.getLogger().println(String.format(MessageConstants.EXIT_EXEC_BAD, DISPLAY_NAME, "Unable to obtain rest client"));
+			return false;
+		}
+
+		IPod p = null;
+		for ( int retries = 10; p == null; retries-- ) {
```

Comment:
I think we should factor in the wait time here vs. simply retries.  And then factor in whatever time we spend here if we have to retry into what is passed into `latch.await(..)`